File: cat.js
const fs = require('fs');
const readline = require('readline');
const path = require('path');

const args = process.argv.slice(2);

const isCopy = args.includes('-c');
const isLine = args.includes('-l');
const filePath = args.find((arg) => !arg.startsWith('-'));

if (!filePath) {
    console.error("no path found ....    ");
    console.log("try cat <filename>");
    console.log("-c for copy file content");
    console.log("-l to show line number");
    process.exit(1);
}

let lineNumber = 1;
let lines = "";

const resolvedPath = path.resolve(filePath);
const rl = readline.createInterface({
    input: fs.createReadStream(resolvedPath, "utf8"),
    output: process.stdout,
    terminal: false
});
rl.on('line', (line) => {
    if (isLine) {
        console.log(`${lineNumber}: ${line}`);
        lineNumber++;
    } else {
        console.log(line);
    }
    if (isCopy) {
        const temp = line + os.EOL;
        lines += temp;
    }
});
rl.on('close', () => {
    if (isCopy) {
        require('child_process').spawn('clip').stdin.end(lines);
        console.log("copied...");
    }
});

File: cat.py
import sys
import subprocess
import os


def copy_to_clipboard(text):
    process = subprocess.Popen('clip', stdin=subprocess.PIPE, shell=True)
    process.communicate(input=text.encode())


def main():
    args = sys.argv[1:]
    lower_args = [arg.lower() for arg in args]

    is_copy = '-c' in lower_args
    is_line = '-l' in lower_args
    filePath = next((arg for arg in args if arg.lower() not in ('-c', '-l')), None)

    if not filePath:
        print("no path found ....    ")
        print("try cat <filename>")
        print("-c for copy file content")
        print("-l to show line number")
        sys.exit(1)

    if not os.path.isfile(filePath):
        print(f"Error: '{filePath}' not found")
        sys.exit(1)

    content = ''
    index = 0
    try:
        with open(filePath, 'r') as file:
            for line in file:
                if is_copy:
                    content += line
                if not is_line:
                    print(line, end='')
                else:
                    index += 1
                    print(f"{index}: {line}", end='')
        if is_copy:
            copy_to_clipboard(content)
    except IOError as e:
        print(f"Error reading file: {e}")
        sys.exit(1)


if __name__ == "__main__":
    main()